### PR TITLE
vm buffered ipld

### PR DESCRIPTION
**Pull request by turuslan**
- enable disabled test-vectors
- fix power v2 claim type
- fix verify post
- fix duplicate error cpp file
- handle exit code
- fix state tree recursion when using dvm
- simplified batch verify seal
- cid utils
- buffered ipld, writes only final changes and not unreferenced temporary changes, also assumes cbor blake cids

### CODE REVIEW

File: core/primitives/cid/cid.hpp [hunk 1]
```diff
@@ -75,6 +79,10 @@ namespace fc {
   };
 
   size_t hash_value(const CID &cid);
+
+  bool isCbor(const CID &cid);
+  boost::optional<Hash256> asBlake(const CID &cid);
+  boost::optional<BytesIn> asIdentity(const CID &cid);
```
Comment (wer1st) [inline core/primitives/cid/cid.hpp:85]:
```suggestion
  bool isCbor(const CID &cid) const;
  boost::optional<Hash256> asBlake(const CID &cid) const;
  boost::optional<BytesIn> asIdentity(const CID &cid) const;
```

File: core/vm/exit_code/impl/exit_code.cpp [hunk 2]
```diff
@@ -35,4 +35,11 @@ namespace fc::vm {
   bool isAbortExitCode(const std::error_code &error) {
     return error.category() == __libp2p::Category<VMAbortExitCode>::get();
   }
+
+  outcome::result<VMExitCode> asExitCode(const std::error_code &error) {
+    if (isVMExitCode(error)) {
+      return outcome::success(VMExitCode{error.value()});
```
Comment (wer1st) [inline core/vm/exit_code/impl/exit_code.cpp:41]:
```suggestion
      return outcome::success(static_cast<VMExitCode>(error.value()));
```

File: core/vm/runtime/impl/runtime_impl.cpp [hunk 3]
```diff
@@ -181,11 +181,37 @@ namespace fc::vm::runtime {
...
+    BatchSealsOut res;
+    if (batch) {
+      OUTCOME_TRY(batch->visit([&](auto &miner,
+                                   auto &seals) -> outcome::result<void> {
+        OUTCOME_TRY(count, seals.size());
+        std::vector<SectorNumber> successful;
+        successful.reserve(count);
+        std::set<SectorNumber> seen;
+        OUTCOME_TRY(seals.visit([&](auto, auto &seal) -> outcome::result<void> {
+          auto verified{proofs::Proofs::verifySeal(seal)};
```
Comment (wer1st) [inline core/vm/runtime/impl/runtime_impl.cpp:201]:
```suggestion
          const auto verified{proofs::Proofs::verifySeal(seal)};
```

File: core/primitives/cid/cid.hpp (hunk 1 above)
Comment (turuslan) [reply core/primitives/cid/cid.hpp:85]: They are not member functions

Comment (wer1st) [reply core/primitives/cid/cid.hpp:85]: Yes, I missed that

**Review by wer1st — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**